src/Controls/src/Core/Layout/Layout.cs

```diff
@@ -9,6 +9,144 @@
 
 namespace Microsoft.Maui.Controls
 {
+
+#pragma warning disable RS0016 // Add public types and members to the declared API
+	public abstract class Layout<T> : Layout where T : View, ICrossPlatformLayout
+	{
+		public new IList<T> Children { get; }
+		public Layout()
+		{
+			Children = new CastingList<T, IView>(this);
+		}
+
+		// If we want to match the APIS
+
+		public void ForceLayout() => SizeAllocated(Width, Height);
+
+		protected override void OnChildAdded(Element child)
+		{
+			base.OnChildAdded(child);
+
+			if (child is T typedChild)
+			{
+				OnAdded(typedChild);
+#pragma warning disable CS0618 // Type or member is obsolete
+				typedChild.MeasureInvalidated += OnChildMeasureInvalidated;
+#pragma warning restore CS0618 // Type or member is obsolete
+			}
+		}
+
+		protected override void OnChildRemoved(Element child, int oldLogicalIndex)
+		{
+			base.OnChildRemoved(child, oldLogicalIndex);
+
+			if (child is T typedChild)
+			{
+				OnRemoved(typedChild);
+#pragma warning disable CS0618 // Type or member is obsolete
+				typedChild.MeasureInvalidated -= OnChildMeasureInvalidated;
+#pragma warning restore CS0618 // Type or member is obsolete
+			}
+		}
+
+		/// <summary>
+		/// Invoked when a child is added to the layout. Implement this method to add class handling for this event.
+		/// </summary>
+		/// <param name="view">The view which was added.</param>
+		protected virtual void OnAdded(T view)
```

Comment:
I know this makes sense to have this, but should we obsolete as well? It is very sad the names do not match the base `OnChildAdded` :(

Maybe we can have 2, this and obsolete and then a new (useful) `OnChildAdded`? Then we can delete the obsolete in net11? Or, this is a small change so we just fix the name now? But both are fine too since the one is obsolete.

Reply:
Yea, I went back and forth on this one a bit

I wasn't sure if having the different typed overrides was confusing or bad. I'm guessing that's why they were added with slightly different names

I went with your suggestion